src/Features/Core/Portable/EventListener/IEventListener`1.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.EventListener
+{
+    /// <summary>
+    /// provide a way for features to lazily subscribe to a service event for particular workspace
+    /// 
+    /// see <see cref="WellKnownEventListeners"/> for supported services
+    /// </summary>
+    internal interface IEventListener<TService> : IEventListener
```

Comment:
If I squint a bit this looks a lot like the IWorkspaceEventListener -- can we simplify and just use this interface where the "service" in question is the workspace itself and has a well known kind of "workspace changes" or something?

Reply:
I merged 2. but these 2 are a similar but subtly different one. so I am not sure whether merging it is better. but not sure whether not merging it is better either so, for now, merged.